src/modules/mavlink/mavlink_messages.cpp

```diff
@@ -5243,6 +5244,172 @@ class MavlinkStreamRawRpm : public MavlinkStream
 	}
 };
 
+class MavlinkStreamESCInfo : public MavlinkStream
+{
+public:
+	const char *get_name() const override
+	{
+		return MavlinkStreamESCInfo::get_name_static();
+	}
+
+	static constexpr const char *get_name_static()
+	{
+		return "ESC_INFO";
+	}
+
+	static constexpr uint16_t get_id_static()
+	{
+		return MAVLINK_MSG_ID_ESC_STATUS;
+	}
+
+	uint16_t get_id() override
+	{
+		return get_id_static();
+	}
+
+	static MavlinkStream *new_instance(Mavlink *mavlink)
+	{
+		return new MavlinkStreamESCInfo(mavlink);
+	}
+
+	unsigned get_size() override
+	{
+		return _esc_info_sub.advertised() ? MAVLINK_MSG_ID_ESC_INFO_LEN + MAVLINK_NUM_NON_PAYLOAD_BYTES : 0;
+	}
+
+private:
+	uORB::Subscription _esc_info_sub{ORB_ID(esc_status)};
```

Comment:
Can you keep the naming in sync? It's already kind of confusing with esc_status and esc_report.
```suggestion
	uORB::Subscription _esc_status_sub{ORB_ID(esc_status)};
```

Reply:
I went with `esc_info` and `esc_status` due to the mavlink message names but yes here the subscription should be `_esc_status_sub` as it's subscribed to the esc_status uORB topic. I'll make the change, but I'm assuming that this is the only name change and that the rest should remain with `esc_info` and `esc_status` depending on the respective mavlink message?

I also agree with you, `esc_status` and `esc_report` is also very confusing. Especially because the esc_status/esc_report messages have a lot of not needed `esc_` prefix that make it even more cumbersome than needed. it def could use an overhaul if there's a better approach/suggestion.